### PR TITLE
#35172 Attempt to resolve UnauthorizedAccessException for Mutex

### PR DESCRIPTION
**Pull request by svick**

### CODE REVIEW

**Review by prochan2 — APPROVED** [review]

**Review by prochan2 — CHANGES_REQUESTED** [review]
There's a copy of this code in DesignTimeEntryPointManager, that needs to be kept in sync.

**Comment by svick** [comment]
@prochan2 Thanks, I missed that. Though the `DesignTimeEntryPointManager` project currently only targets .Net Standard and .Net 6, so I had to change that.

File: Metalama.Backstage/Utilities/MutexHelper.cs [hunk 1]
```diff
@@ -108,21 +110,32 @@ internal static Mutex OpenOrCreateMutex( string mutexName, ILogger? logger )
                 // Otherwise we will try to create the mutex.
                 try
                 {
-                    logger?.Trace?.Log( $"  Creating new mutex." );
+                    logger?.Trace?.Log( "  Creating new mutex." );
```
Comment (prochan2) [inline Metalama.Backstage/Utilities/MutexHelper.cs:113]:
```suggestion
```

File: Metalama.Backstage/Utilities/MutexHelper.cs [hunk 2]
```diff
@@ -108,21 +110,32 @@ internal static Mutex OpenOrCreateMutex( string mutexName, ILogger? logger )
...
+
+#if NETFRAMEWORK
+                    // From https://stackoverflow.com/a/19717341/41071.
+                    // As I understand it, creating a mutex without security descriptor uses default security, which could be different on different systems.
+                    // I'm not certain this will actually prevent UnauthorizedAccessException, and it's not available on .NET, but it's worth trying.
 
+                    var mutexSecurity = new MutexSecurity();
```
Comment (prochan2) [inline Metalama.Backstage/Utilities/MutexHelper.cs:120]:
```suggestion
                    logger?.Trace?.Log( "  Creating new mutex with access rule." );
                    var mutexSecurity = new MutexSecurity();
```

File: Metalama.Backstage/Utilities/MutexHelper.cs [hunk 3]
```diff
@@ -108,21 +110,32 @@ internal static Mutex OpenOrCreateMutex( string mutexName, ILogger? logger )
...
+                    mutexSecurity.AddAccessRule( new MutexAccessRule( new SecurityIdentifier( WellKnownSidType.WorldSid, null ), MutexRights.Synchronize | MutexRights.Modify, AccessControlType.Allow ) );
+
+                    return new Mutex( false, mutexName, out _, mutexSecurity );
+#else
                     return new Mutex( false, mutexName );
```
Comment (prochan2) [inline Metalama.Backstage/Utilities/MutexHelper.cs:125]:
```suggestion
                    logger?.Trace?.Log( "  Creating new mutex." );
                    return new Mutex( false, mutexName );
```

**Review by prochan2 — APPROVED** [review]

**Pull request merged**